backend/src/api/database-migration.ts

```diff
@@ -0,0 +1,176 @@
+import config from '../config';
+import { DB } from '../database';
+import logger from '../logger';
+
+class DatabaseMigration {
+  private static currentVersion = 1;
+  private queryTimeout = 120000;
+
+  constructor() { }
+
+  public async $initializeOrMigrateDatabase(): Promise<void> {
+    if (!await this.$checkIfTableExists('statistics')) {
+      const connection = await DB.pool.getConnection();
+      for (const query of this.getInitializeTableQueries()) {
+        await connection.query<any>({ sql: query, timeout: this.queryTimeout });
+      }
+      connection.release();
+      logger.info(`Initial database tables has been created`);
+    }
+
+    if (await this.$checkIfTableExists('state')) {
+      const databaseSchemaVersion = await this.$getSchemaVersionFromDatabase();
+      if (DatabaseMigration.currentVersion > databaseSchemaVersion) {
+        const connection = await DB.pool.getConnection();
+        for (const query of this.getMigrationQueriesFromVersion(databaseSchemaVersion)) {
+          await connection.query<any>({ sql: query, timeout: this.queryTimeout });
+        }
+        connection.release();
+        this.$updateToLatestSchemaVersion();
+        logger.info(`Database schema has been migrated to the latest version (${DatabaseMigration.currentVersion})`);
```

Comment:
```suggestion
        logger.info(`Database schema has been migrated from ${databaseSchemaVersion} to the latest version: (${DatabaseMigration.currentVersion})`);
```

Reply:
@softsimon I think it's important to log what version we are migrating from as well